lib/sdr_client/redesigned_client.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require 'logger'
+require 'singleton'
+
+module SdrClient
+  # The SDR client reimagined, built using patterns successfully used in other client gems we maintain
+  class RedesignedClient
+    include Singleton
+
+    DEFAULT_HEADERS = {
+      accept: 'application/json',
+      content_type: 'application/json'
+    }.freeze
+
+    class << self
+      def configure(url:, email:, password:, request_options: default_request_options, logger: default_logger)
+        instance.config = Config.new(
+          url: url,
+          email: email,
+          password: password,
+          request_options: request_options,
+          logger: logger
+        )
+
+        instance
+      end
+
+      def default_logger
+        Logger.new($stdout)
+      end
+
+      def default_request_options
+        {
+          read_timeout: default_timeout,
+          timeout: default_timeout
+        }
+      end
+
+      # NOTE: This is the number of seconds it roughly takes for H2 to
+      #       successfully shunt ~10GB files over to SDR API
+      def default_timeout
+        900
+      end
+
+      delegate :config, :connection, :deposit_model, :job_status, to: :instance
+    end
+
+    attr_accessor :config
+
+    def deposit_model(...)
+      Deposit.deposit_model(...)
+    end
+
+    def job_status(...)
+      JobStatus.new(...)
+    end
+
+    # def token_for(user_id:)
+    # end
+
+    # def checksum_for(file_path:, algorithm: :md5|:sha1)
+    # end
+
+    # def find(object_id:)
+    # end
+
+    # def update_model(model:, version_description:)
+    # end
+
+    # # NOTE: this method is used only the CLI & the infrastructure-integration-tests deposit helpers
+    # def deposit_metadata(apo:, collection:, type:, strategies:, files:, path:)
+    # end
```

Comment:
These commented out methods *should* map to the rest of the interface we need for H2, Argo, and the integration test helpers, but I'm focusing on google-books to start.